build-source-tarball.sh

```diff
@@ -146,8 +147,14 @@ git submodule foreach --quiet --recursive '
 
 # Now re-uberclone into the tarball src directory.  Since we reuse the .gitdirs, this shouldn't hit the network at all.
 ignored_repos="https://dev.azure.com/dnceng/internal/_git/dotnet-optimization;https://dev.azure.com/devdiv/DevDiv/_git/DotNet-Trusted;https://devdiv.visualstudio.com/DevDiv/_git/DotNet-Trusted;https://dnceng@dev.azure.com/dnceng/internal/_git/dotnet-optimization;https://dev.azure.com/dnceng/internal/_git/dotnet-core-setup;https://github.com/dotnet/source-build-reference-packages"
+
+#export the LC_LIB_PATH for libgit2 so file as fedora fails to find it in the repodir
+export LD_LIBRARY_PATH=$CLI_PATH/tools/.store/microsoft.dotnet.darc/$DarcVersion/microsoft.dotnet.darc/$DarcVersion/tools/netcoreapp2.1/any/runtimes/rhel-x64/native/
```

Comment:
It might be worth opening an issue in LibGit2Sharp for this too to figure out what the actual cause is.

Reply:
Done - [#1730](https://github.com/libgit2/libgit2sharp/issues/1730)

Reply:
FYI, we ran into this on Fedora 30 and RHEL 8 on arm64 as well. @tmds fixed it with https://github.com/libgit2/libgit2sharp/pull/1714.

I think we should remove this `LD_LIBRARY_PATH` change (which is not portable) and bump libgit2sharp 0.27.0-preview-0020 to pick up https://github.com/libgit2/libgit2sharp/pull/1714 which is more portable. I also filed https://github.com/dotnet/source-build/issues/1354 to fix this same underlying issue.

Reply:
To follow up a bit more, it seems like this is a functional change caused because we are now downloading the darc built against .NET Core 2.1 instead of compiling one using .NET Core 3.0. darc (actually libgit2sharp) took advantage of new features in .NET Core 3.0 which meant that this LD_LIBRARY_PATH change was not needed.